lib/client.js

```
@@ -92,6 +141,243 @@ function MatrixClient(opts) {
+ */
+MatrixClient.prototype.setRoomEncryption = function(roomId, config) {
+    if (config.algorithm === OLM_ALGORITHM) {
+        if (!config.members) {
+            throw new Error(
+                "Config must include a 'members' list with a list of userIds"
+            );
+        }
+        var devicesWithoutSession = [];
+        var userWithoutDevices = [];
+        for (var i = 0; i < config.members.length; ++i) {
+            var userId = config.members[i];
+            var devices = this.sessionStore.getEndToEndDevicesForUser(userId);
+            if (!devices) {
+                userWithoutDevices.push(userId);
+            } else {
+                for (var deviceId in devices) {
+                    if (devices.hasOwnProperty(deviceId)) {
+                        var keys = devices[deviceId];
+                        var key = keys.keys["curve25519:" + deviceId];
+                        if (key == this.deviceCurve25519Key) {
+                            continue;
+                        }
+                        if (!this.sessionStore.getEndToEndSessions(key)) {
+                            devicesWithoutSession.push([userId, deviceId, key]);
+                        }
+                    }
+                }
+            }
+        }
+        var deferred = q.defer();
+        if (devicesWithoutSession.length > 0) {
+            var queries = {};
+            for (i = 0; i < devicesWithoutSession.length; ++i) {
+                var device = devicesWithoutSession[i];
+                var query = queries[device[0]] || {};
+                queries[device[0]] = query;
+                query[device[1]] = "curve25519";
+            }
+            var path = "/keys/claim";
+            var content = {one_time_keys: queries};
+            var self = this;
+            this._http.authedRequestWithPrefix(
+                undefined, "POST", path, undefined, content,
+                httpApi.PREFIX_V2_ALPHA
+            ).then(function(res) {
```

think this wants to be .done to avoid swallowing exceptions